src/engine/Scene.ts

```diff
@@ -307,9 +307,8 @@ export class Scene extends Class implements ICanInitialize, ICanActivate, ICanDe
     */
    public update(engine: Engine, delta: number) {
       this._preupdate(engine, delta);
-      var i: number, len: number;
-
-      
+      let i: number;
```

Comment:
While I agree with these changes in principle, lets keep this PR focused on the issue at hand and not modify things that aren't part of event propagation.